From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 24 Nov 2011 07:45:52 +0000
Subject: [PATCH] bcma: Do not claim PCI device IDs also claimed by brcmsmac
Forwarded: not-needed

brcmsmac was not updated to use the bcma bus driver until Linux 3.3.
For 3.2 we want bcma to claim only the devices that brcmsmac doesn't
support.

Signed-off-by: Ben Hutchings <ben@decadent.org.uk>
---
 drivers/bcma/host_pci.c                |    4 +++-
 drivers/net/wireless/brcm80211/Kconfig |    1 -
 2 files changed, 3 insertions(+), 2 deletions(-)

--- a/drivers/bcma/host_pci.c
+++ b/drivers/bcma/host_pci.c
@@ -263,11 +263,13 @@
 #endif /* CONFIG_PM */
 
 static DEFINE_PCI_DEVICE_TABLE(bcma_pci_bridge_tbl) = {
+#if !IS_ENABLED(CONFIG_BRCMSMAC)
 	{ PCI_DEVICE(PCI_VENDOR_ID_BROADCOM, 0x0576) },
-	{ PCI_DEVICE(PCI_VENDOR_ID_BROADCOM, 0x4331) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_BROADCOM, 0x4353) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_BROADCOM, 0x4357) },
 	{ PCI_DEVICE(PCI_VENDOR_ID_BROADCOM, 0x4727) },
+#endif
+	{ PCI_DEVICE(PCI_VENDOR_ID_BROADCOM, 0x4331) },
 	{ 0, },
 };
 MODULE_DEVICE_TABLE(pci, bcma_pci_bridge_tbl);
--- a/drivers/net/wireless/brcm80211/Kconfig
+++ b/drivers/net/wireless/brcm80211/Kconfig
@@ -5,7 +5,6 @@
 	tristate "Broadcom IEEE802.11n PCIe SoftMAC WLAN driver"
 	depends on PCI
 	depends on MAC80211
-	depends on BCMA=n
 	select BRCMUTIL
 	select FW_LOADER
 	select CRC_CCITT
